runtime/src/main/java/org/corfudb/runtime/GarbageInformer.java

```diff
@@ -86,14 +106,23 @@ public void start() {
                 GC_PERIOD.getSeconds() + rand.nextInt((int) GC_PERIOD.getSeconds()),
                 GC_PERIOD.getSeconds(),
                 TimeUnit.SECONDS);
+
+        log.info("Garbage informer is started.");
     }
 
     /**
      * Stop to send garbage decisions to LogUnit servers.
      * This function shuts down the single-threaded executor.
      */
     public void stop() {
+        if (!isRunning.get()) {
```

Comment:
I think it's better to use compareAndSet just as what you did in start()

Reply:
Let's discuss it offline.